src/test/java/api/InstancesApiExamples.java

```diff
@@ -419,6 +417,141 @@ public void cannotUpdateAnInstanceThatDoesNotExist()
     assertThat(putResponse.getStatusCode(), is(404));
   }
 
+  @Test
+  public void canUpdateAnExistingMARCInstanceIfNoChanges()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    UUID id = UUID.randomUUID();
+    // Create new Instance
+    JsonObject newInstance = createInstance(treasureIslandInstance(id));
+    JsonObject instanceForUpdate = treasureIslandInstance(id);
+    URL instanceLocation = new URL(String.format("%s/%s", ApiRoot.instances(), newInstance.getString("id")));
+    // Put Instance for update
+    CompletableFuture<Response> putCompleted = new CompletableFuture<>();
+    okapiClient.put(instanceLocation, instanceForUpdate, ResponseHandler.any(putCompleted));
+    Response putResponse = putCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(putResponse.getStatusCode(), is(204));
+    // Get existing Instance
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
+    okapiClient.get(instanceLocation, ResponseHandler.json(getCompleted));
+    Response getResponse = getCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(getResponse.getStatusCode(), is(200));
+
+    JsonObject updatedInstance = getResponse.getJson();
+    assertEquals(updatedInstance, newInstance);
+    selfLinkRespectsWayResourceWasReached(updatedInstance);
+    selfLinkShouldBeReachable(updatedInstance);
+  }
+
+  @Test
+  public void canNotUpdateAnExistingMARCInstanceIfBlockedFieldsAreChanged()
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    UUID id = UUID.randomUUID();
+    createInstance(treasureIslandInstance(id));
+    JsonObject instanceForUpdate = marcInstanceWithDefaultBlockedFields(id);
+
+    for (String field : config.getInstanceBlockedFields()) {
+      URL instanceLocation = new URL(String.format("%s/%s", ApiRoot.instances(), id));
+      // Put Instance for update
+      CompletableFuture<Response> putCompleted = new CompletableFuture<>();
+      okapiClient.put(instanceLocation, instanceForUpdate, ResponseHandler.any(putCompleted));
+      Response putResponse = putCompleted.get(5, TimeUnit.SECONDS);
+
+      assertThat(putResponse.getStatusCode(), is(422));
+      assertThat(putResponse.getJson().getJsonArray("errors").size(), is(1));
```

Comment:
Could it be worthwhile to add checks that the validation error is the one that is expected?

Does the implementation accumulate errors for each blocked property that is different?

Reply:
I think -yes, It is worth to add a check for the whole error message.
The implementation does not accumulate errors and differences - this implementation may be called only if the client sends request bypassing the frontend. The work to keep fields in a proper unchanged values lies on a client is this case.

Reply:
I added a check for the error message in this test method `canNotUpdateAnExistingMARCInstanceIfBlockedFieldsAreChangedToNulls`